pkg/genlib/generator_interface.go

```diff
@@ -0,0 +1,289 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
here we moved all the common code regardless template usage that was in `generator.go`